pyproject.toml

```diff
@@ -114,6 +114,7 @@ remote_data_strict = true
 filterwarnings = [
     'error',
     'ignore:numpy.ndarray size changed:RuntimeWarning',
+    'ignore:.*extensions.*deprecated:asdf.exceptions.AsdfDeprecationWarning',
```

Comment:
Presumably this is needed in part because the test code uses the deprecated feature. Here's one example:
https://github.com/asdf-format/asdf/blob/19d2cbb4486559cd5b44fa400823456d9f6bd05c/asdf/_tests/test_schema.py#L307

As these deprecated uses will need to be fixed when the feature is removed it seems preferable to make these changes as part of this PR and to remove this general warning filter.